File: app/src/main/java/com/shevelev/comics_viewer/activities/main/comics_filters/ComicsFilterFactory.java
package com.shevelev.comics_viewer.activities.main.comics_filters;

import com.shevelev.comics_viewer.activities.main.ComicsSortInfo;
import com.shevelev.comics_viewer.options.OptionsFacade;
import com.shevelev.comics_viewer.options.OptionsKeys;

public class ComicsFilterFactory
{
    public static IComicsFilter getFilter(ComicsViewMode mode)
    {
        boolean isPrivateComicsHidden = OptionsFacade.ShortLivings.get(OptionsKeys.PasswordEntered)==null;

        if(mode== ComicsViewMode.All)
            return new AllComicsFilter(new ComicsSortInfo((c1, c2) -> c1.name.compareTo(c2.name), false), isPrivateComicsHidden);
        else
            return new RecentComicsFilter(null, isPrivateComicsHidden);
    }
}

File: app/src/main/java/com/shevelev/comics_viewer/activities/main/comics_filters/ComicsFilterFactory.kt
package com.shevelev.comics_viewer.activities.main.comics_filters

import com.shevelev.comics_viewer.activities.main.ComicsSortInfo
import com.shevelev.comics_viewer.dal.dto.Comics
import com.shevelev.comics_viewer.options.OptionsFacade
import com.shevelev.comics_viewer.options.OptionsKeys

object ComicsFilterFactory {
    fun getFilter(mode: ComicsViewMode): IComicsFilter {
        val isPrivateComicsHidden = OptionsFacade.ShortLivings[OptionsKeys.PasswordEntered] == null
        return if (mode == ComicsViewMode.ALL) {
            AllComicsFilter(ComicsSortInfo(java.util.Comparator { c1: Comics, c2: Comics -> c1.name.compareTo(c2.name) }, false), isPrivateComicsHidden)
        }
        else {
            RecentComicsFilter(null, isPrivateComicsHidden)
        }
    }
}